docs/api.md

```diff
@@ -2097,7 +2097,7 @@ To pass arguments from node.js to the predicate of `page.waitForFunction` functi
 
 ```js
 const selector = '.foo';
-await page.waitForFunction(selector => !!document.querySelector(selector), {}, selector);
+await page.waitForFunction(async selector => !!document.querySelector(selector), {}, selector);
```

Comment:
This example does not need `async`, so let's leave it as-is and instead add an additional example that does something asynchronously.

Reply:
@mathiasbynens what do you think of this example : 

`page.waitForFunction` also works with asynchronous promises:

```js
const username = 'github-username';
await page.waitForFunction(async username => {
  let githubResponse = await fetch(`https://api.github.com/users/${username}`);
  let githubUser = await githubResponse.json();
  // show the avatar
  let img = document.createElement('img');
  img.src = githubUser.avatar_url;
  // wait 3 seconds
  await new Promise((resolve, reject) => setTimeout(resolve, 3000));
  img.remove();
}, {}, username);
```

Reply:
Looks good, but let's use `const` instead of `let` to match the Puppeteer coding style.

Reply:
@mathiasbynens Done 😄